sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -307,6 +303,151 @@ export class ApplicationService extends RecordDataModelService<Application> {
+      studentId,
+      ApplicationStatus.Completed,
+      applicationId,
+    );
+    // If the application was not found it is because it does not belong to the student or it is not completed.
+    if (!application) {
+      throw new CustomNamedError(
+        "Student application not found or it is not in the correct status to be changed.",
+        APPLICATION_NOT_FOUND,
+      );
+    }
+    if (!allowApplicationChangeRequest(application.programYear)) {
+      throw new CustomNamedError(
+        "The program year associated to this application does not allow a change request submission.",
+        APPLICATION_NOT_VALID,
+      );
+    }
+    // Check if there is already a change request in progress.
+    const hasChangeInProgress = application.parentApplication.versions.some(
+      (version) =>
+        [
+          ApplicationEditStatus.ChangeInProgress,
+          ApplicationEditStatus.ChangePendingApproval,
+        ].includes(version.applicationEditStatus),
+    );
+    if (hasChangeInProgress) {
+      throw new CustomNamedError(
+        "An application change request is already in progress.",
+        APPLICATION_CHANGE_REQUEST_ALREADY_IN_PROGRESS,
+      );
+    }
+    // Validates selected location and selected offering as key values that
+    // should not be changed. Even if some other values are changed (tampered by the user),
+    // there will be no impact for application processing since the location and offering
+    // are copied from the current application retrieved from the database.
+    if (
+      applicationData.selectedLocation !== application.data.selectedLocation
+    ) {
+      throw new CustomNamedError(
+        "Change request has a different location from its original submission.",
+        APPLICATION_NOT_VALID,
+      );
+    }
+    if (
+      applicationData.selectedOffering !== application.data.selectedOffering
+    ) {
+      // Applications are expected to have the same offering value populated, even for PIRs.
+      // PIRs will not have the offering value populated and the assessment will be created
+      // based on the most recent offering associated with the current application version.
+      throw new CustomNamedError(
+        "Change request has a different offering from its original submission.",
+        APPLICATION_NOT_VALID,
+      );
+    }
+
+    // Create the new assessment.
+    const now = new Date();
+    const auditUser = { id: auditUserId } as User;
+    const originalAssessment = new StudentAssessment();
+    originalAssessment.triggerType = AssessmentTriggerType.OriginalAssessment;
+    originalAssessment.submittedBy = auditUser;
+    originalAssessment.submittedDate = now;
+    originalAssessment.creator = auditUser;
+    originalAssessment.offering = {
+      id: application.currentAssessment.offering.id,
+    } as EducationProgramOffering;
+    if (application.currentAssessment.studentAppeal) {
+      originalAssessment.studentAppeal = {
```

Comment:
what about `originalAssessment.studentAppeal = application.currentAssessment.studentAppeal `?

Reply:
It was the initial thought and than I changed relying on the fact that the other references were created using only the `id`, which seems safer then realying on the fact the query will return a more raw `studentAppeal` which would contains only the id and nothing else selected in a way that could interfere with the saving operation.